src/distributors/adoptopenjdk-installer.ts

```diff
@@ -0,0 +1,80 @@
+import * as core from '@actions/core';
+import * as tc from '@actions/tool-cache';
+
+import fs from 'fs';
+import path from 'path';
+import semver from 'semver';
+
+import { IS_WINDOWS, PLATFORM, IS_MACOS, macOSJavaContentDir } from "../util";
+import { JavaInstallerResults, JavaDownloadRelease, JavaBase, JavaInstallerOptions } from "./base-installer";
+import { IRelease, IReleaseVersion } from './adoptopenjdk-models'
+
+export class AdoptOpenJDKDistributor extends JavaBase {
+    
+    constructor(installerOptions: JavaInstallerOptions) {
+        super("AdoptOpenJDK", installerOptions);
+    }
+
+    protected async findPackageForDownload(version: semver.Range): Promise<JavaDownloadRelease> {
+        const platform = IS_MACOS ? 'mac' : PLATFORM;
+
+        const resolvedMajorVersion = await this.resolveMajorVersion(version);
+        const availableVersionsUrl = `https://api.adoptopenjdk.net/v3/assets/feature_releases/${resolvedMajorVersion}/ga?heap_size=normal&image_type=${this.javaPackage}&page=0&page_size=1000&project=jdk&sort_method=DEFAULT&sort_order=DESC&vendor=adoptopenjdk&jvm_impl=hotspot&architecture=${this.arch}&os=${platform}`;
+        const availableVersionsList = ( await this.http.getJson<IRelease[]>(availableVersionsUrl)).result;
+        const resolvedFullVersion = availableVersionsList?.find(item => semver.satisfies(item.version_data.semver, version));
+
+        if(!resolvedFullVersion) {
+            const availableOptions = availableVersionsList?.map(item => item.version_data.semver).join(', ');
+            const availableOptionsMessage = availableOptions ? `\nAvailable versions: ${availableOptions}` : "";
+            throw new Error(`Could not find satisfied version for semver ${version.raw}. ${availableOptionsMessage}`);
+        }
+
+        return {
+            resolvedVersion: resolvedFullVersion.version_data.semver,
+            link: resolvedFullVersion.binaries[0].package.link
```

Comment:
Can `resolvedFullVersion.binaries` contains more than 0 elements? Why we always choose the first one?
Can it also be empty?

Reply:
Done. Thank you.